ecr_helper/helper.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+#
+# Copyright 2018 Google, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+##
+## Validate environment.
+##
+
+checkBinary() {
+    if ! which $1 >&/dev/null; then
+        echo "Unable to locate $1, please ensure it is installed and on your \$PATH."
+        exit 1
+    fi
+}
+
+checkBinary aws
+checkBinary jq
+checkBinary kubectl
+
+readonly KUBECTL_FLAGS="${1:+ -n $1}"
+
+if ! kubectl $KUBECTL_FLAGS get sa >& /dev/null; then
+    echo "Unable to read Kubernetes service accounts with 'kubectl $KUBECTL_FLAGS get sa'."
+    exit 1
+fi
+
+readonly KUBE_SA=${2:-"builder"}
+
+
+##
+## Begin doing things
+##
+
+# Supress stderr, as many of the check queries will print extra output
+# if the resources are not present. Keep stderr on FD 3 to allow
+# printing output from explicit create commands.
+exec 3>&2
+exec 2>/dev/null
+
+# The token expires in 12 hours, as of October, 2018.
+DATA=$(aws ecr get-authorization-token)
```

Comment:
This call will likely fail if the user forgot to run aws configure or attached an incorrect role. It looks like this will result in empty fields being passed through to kubectl where it will eventually fail for nil username and password. I would recommend adding an explicit exit code check here to make debugging easier.

Reply:
added check